App.Goods/Services/ValidateService.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using App.Configuration;
+using App.Goods.Common;
+
+namespace App.Goods.Services
+{
+    public class ValidateService : IValidateService, ITransientDependency
+    {
+        private readonly IProductsManager _productsManager;
+
+        public ValidateService(IProductsManager productsManager)
+        {
+            _productsManager = productsManager;
+        }
+
+        public void ValidateIds(List<int> ids)
```

Comment:
Method name and what it actually did is not correspondent. In most cases "Validators" checks model or some input for validity and then return the validity state somehow, either return boolean or throwing exception.
In your case method is not doing validation, it is doing cleaning. So the valid name should be "Cleaner".
Or do as I wrote before - throw exception when receiving unexpected productId